combat-trainer.lic

```diff
@@ -55,12 +43,12 @@ class SetupProcess
       game_state.next_clean_up_step
       return true
     end
-    game_state.dancing = should_dance?
-    game_state.now_retreating = should_retreat?
+    game_state.update_dancing(DRRoom.npcs.length)
+    game_state.update_retreating(DRRoom.npcs.length)
     if game_state.dancing
-      dance(game_state)
+      game_state.dance(@weapon_training)
```

Comment:
From reading this line I really don't have any good idea what this call would do.